extras/api/views.py

```diff
@@ -181,3 +191,195 @@ class WebhookViewSet(PeeringManagerModelViewSet):
     queryset = Webhook.objects.all()
     serializer_class = WebhookSerializer
     filterset_class = WebhookFilterSet
+
+
+class BGPq4ViewSet(ViewSet):
+    """
+    ViewSet for BGPq4 prefix queries using existing Redis cache and BGPq infrastructure
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get_redis_client(self):
+        """Get Redis client using existing Django cache configuration"""
+
+        # Get the default cache backend
+        cache = caches["default"]
+
+        # For django-redis, access the client directly
+        if hasattr(cache, "client"):
+            return cache.client.get_client()
+
+        # Fallback: create Redis client from Django settings
+        import redis
+        from django.conf import settings
+
+        # Assuming CACHES config has Redis URL or connection params
+        cache_config = settings.CACHES.get("default", {})
+        location = cache_config.get("LOCATION", "redis://redis:6379/0")
+
+        return redis.from_url(location)
```

Comment:
There is no need to build a redis client to access the cache. Django can provide us with one by having the following import `from django.core.cache import cache`.